src/dbt_client/dbtCloudIntegration.ts

```diff
@@ -724,10 +726,71 @@ export class DBTCloudProjectIntegration
     );
   }
 
+  async getBulkCompiledSQL(models: NodeMetaData[]) {
+    const downloadArtifactsVersion = "0.37.20";
+    const currentVersion = this.getVersion()
+      .map((part) => new String(part))
+      .join(".");
+    if (semver.gte(currentVersion, downloadArtifactsVersion)) {
+      const compileQueryCommand = this.dbtCloudCommand(
+        new DBTCommand("Getting catalog...", [
+          "compile",
+          "--download-artifacts",
+          "--model",
+          '"' + models.map((item) => item.name).join(" ") + '"',
+          "--output",
+          "json",
+          "--log-format",
+          "json",
+        ]),
+      );
+      const { stderr } = await compileQueryCommand.execute(
+        new CancellationTokenSource().token,
+      );
+      const exception = this.processJSONErrors(stderr);
```

Comment:
is this method required to return anything? the else case returns result, but nothing here

Reply:
no. this case basically generate artifacts in the filesystem, which is then read below the if block